tools/commit_slider/perf_cross_check.md

```diff
@@ -0,0 +1,336 @@
+    self.firstAppCmd = cfg["runConfig"]["firstAppCmd"]
+    self.secondAppCmd = cfg["runConfig"]["secondAppCmd"]
+    self.outPattern = r'{spec}:\s*([0-9]*[.][0-9]*)\s*{measure}'.format(spec='Throughput', measure='FPS')
+    super().checkCfg(cfg)
+
+def  getPseudoMetric(self, commit, cfg):
+    # main Mode method initially resolving two compared
+    # commits (for performance goals it is numeric metric, as
+    # soon as in some Modes 'metric' has non-numeric value,
+    # e.g. filename we got pseudometric), for noncomparative
+    # Modes it doesn't return any value, just run procedures
+    # for handling single commit.
+    commit = commit.replace('"', "")
+    commitLogger = getCommitLogger(cfg, commit)
+    self.commonLogger.info("New commit: {commit}".format(commit=commit))
+
+    # building and compilation of given version
+    handleCommit(commit, cfg)
+    fullOutput = ""
+
+    # passing each of the commandlines for execution
+    simpleSubstitute(cfg, "actualAppCmd", "$.runConfig.firstAppCmd", "$.appCmd")
+
+    # run first app
+    checkOut = fetchAppOutput(cfg, commit)
+    commitLogger.info(checkOut)
+    foundThroughput = re.search(
+        self.outPattern, checkOut, flags=re.MULTILINE
+    ).group(1)
+
+    # parsing of output and holding it with full commandline
+    # and specific model for the final result
+    self.firstThroughput = foundThroughput
+    self.firstModel = cfg['appCmd']
+    fullOutput = checkOut
+    simpleSubstitute(cfg, "actualAppCmd", "$.runConfig.secondAppCmd", "$.appCmd")
+
+    # run second app
+    checkOut = fetchAppOutput(cfg, commit)
+    commitLogger.info(checkOut)
+    foundThroughput = re.search(
+        self.outPattern, checkOut, flags=re.MULTILINE
+    ).group(1)
+    self.secondThroughput = foundThroughput
+    self.secondModel = cfg['appCmd']
+
+    # add commit data to commit path
+    pc = Mode.CommitPath.PathCommit(
+        commit,
+        Mode.CommitPath.CommitState.DEFAULT
+    )
+    self.setOutputInfo(pc)
+    self.commitPath.accept(self.traversal, pc)
+
+def  printResult(self):
+    # formatting of full output
+    for  pathcommit  in  self.commitPath.getList():
+        commitInfo = self.getCommitInfo(pathcommit)
+        print(commitInfo)
+        self.outLogger.info(commitInfo)
+
+def  setOutputInfo(self, pathCommit):
+    # populating of commit path
+    pathCommit.firstThroughput = self.firstThroughput
+    pathCommit.firstModel = self.firstModel
+    pathCommit.secondThroughput = self.secondThroughput
+    pathCommit.secondModel = self.secondModel
+
+def  getCommitInfo(self, commit):
+    return  "{hash}, throughput_1 = {t1}, throughput_2 = {t2}".format(
+        hash=commit.cHash,
+        t1=commit.firstThroughput,
+        t2=commit.secondThroughput)
+```
+### Configuration
+```
+supposedly we pass both commandlines launching models in two parameters cfg["runConfig"]["firstAppCmd"] and cfg["runConfig"]["secondAppCmd"], i.e.
```

Comment:
Should it be inside code block? If yes, can we place on several lines to avoid horizontal scrolling?